consensus/scheme/rolldpos/rolldposctx.go

```diff
@@ -405,6 +406,38 @@ func (ctx *rollDPoSCtx) Proposal() (interface{}, error) {
 	return ctx.mintNewBlock(privateKey)
 }
 
+func (ctx *rollDPoSCtx) prepareNextProposal(prevHeight uint64, prevHash hash.Hash256) error {
+	var (
+		height    = prevHeight + 1
+		interval  = ctx.BlockInterval(height)
+		startTime = ctx.round.StartTime().Add(interval)
+		err       error
+	)
+	fork, err := ctx.chain.Fork(prevHash)
+	if err != nil {
+		return errors.Wrapf(err, "failed to check fork at block %d, hash %x", prevHeight, prevHash[:])
+	}
+	roundCalc := ctx.roundCalc.Fork(fork)
+	// check if the current node is the next proposer
+	nextProposer := roundCalc.Proposer(height, interval, startTime)
+	var privateKey crypto.PrivateKey = nil
+	if idx := slices.Index(ctx.encodedAddrs, nextProposer); idx < 0 {
+		return nil
+	} else {
+		privateKey = ctx.priKeys[idx]
+	}
+	ctx.logger().Debug("prepare next proposal", log.Hex("prevHash", prevHash[:]), zap.Uint64("height", ctx.round.height+1), zap.Time("timestamp", startTime), zap.String("nextproposer", nextProposer))
+	go func() {
```

Comment:
why use `go func()` here

Reply:
minor:
if fork.MintNewBlock() errs out, the error won't be captured and handled

Reply:
`MintNewBlock` keep mint result in memory, and will return it if mint same block. Here is to just to start the next proposal mint in advance, which can be run in background.